spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDeploymentController.java

```diff
@@ -63,11 +65,13 @@ public TaskDeploymentController(TaskService taskService) {
 	 * @param taskName the name of the existing task to be executed (required)
 	 * @param properties the runtime properties for the task, as a comma-delimited list of
 	 * 					 key=value pairs
+	 * @param params the runtime commandline arguments
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.POST)
 	@ResponseStatus(HttpStatus.CREATED)
-	public void deploy(@PathVariable("name") String taskName, @RequestParam(required = false) String properties) {
-		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties));
+	public void deploy(@PathVariable("name") String taskName, @RequestParam(required = false) String properties,
+			@RequestParam(required = false) List<String> params) {
+		this.taskService.executeTask(taskName, DeploymentPropertiesUtils.parse(properties), params);
```

Comment:
I know there is a better way to do this but:
```
		Pattern regex = Pattern.compile("(\\s(?=([^\\\"']*[\\\"'][^\\\"']*[\\\"'])*[^\\\"']*$))");
		for( String paramsItem : params) {
			Matcher regexMatcher = regex.matcher(paramsItem);
			int start = 0;
			while (regexMatcher.find()) {
				paramsToUse.add(paramsItem.substring(start, regexMatcher.start()));
				start = regexMatcher.start();
			}
			if(paramsItem != null && paramsItem.length() > 0){
				paramsToUse.add(paramsItem.substring(start,paramsItem.length()));
			}
		}```
Launching this task:   task launch foo --params "--format=\"yyyy-MM-dd HH:mm:ss.SSS\" bifflediffle=snizersnot wombats=\"eat veggies\" noway=really"

displayed this in my sample task: 
 you got ==>--format="yyyy-MM-dd HH:mm:ss.SSS"
 you got ==> bifflediffle=snizersnot
 you got ==> wombats="eat veggies"
 you got ==> noway=really